packages/meditrak-app/app/assessment/selectors.js

```diff
@@ -109,3 +110,85 @@ export const getQuestion = (state, questionId) => state.assessment.questions[que
 export const getAnswers = state => state.assessment.answers;
 
 export const getAnswerForQuestion = (state, questionId) => getAnswers(state)[questionId];
+
+export const getArithmeticResult = (state, arithmeticQuestionId) => {
+  const { config } = getQuestion(state, arithmeticQuestionId);
+
+  const {
+    arithmetic: { formula, valueTranslation = {}, defaultValues = {}, answerDisplayText = '' },
+  } = config;
+
+  const values = {};
+  const expressionParser = new ExpressionParser();
+  const variables = expressionParser.getVariables(formula);
+  let translatedAnswerDisplayText = answerDisplayText;
+
+  const getArithmeticAnswer = questionId => {
+    const answer = getAnswerForQuestion(state, questionId);
+
+    if (valueTranslation[questionId] && valueTranslation[questionId][answer] !== undefined) {
+      return valueTranslation[questionId][answer]; // return translated answer if there's any
+    }
+
+    if (answer !== undefined) {
+      return answer; // return raw answer
+    }
+
+    return defaultValues[questionId] !== undefined ? defaultValues[questionId] : 0; // No answer found, return the default answer
+  };
+
+  // Setting up scope values.
+  variables.forEach(questionIdVariable => {
+    const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix to get the actual questionId
+    values[questionIdVariable] = getArithmeticAnswer(questionId);
+  });
+
+  // Replace variables with real values in answerDisplayText
+  variables.forEach(questionIdVariable => {
+    const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix to get the actual questionId
+    const answer = getArithmeticAnswer(questionId);
+    translatedAnswerDisplayText = translatedAnswerDisplayText.replace(questionId, answer);
+  });
+
+  expressionParser.setScope(values);
+
+  const result = !isNaN(expressionParser.evaluate(formula))
+    ? Math.round(expressionParser.evaluate(formula) * 1000) / 1000 // Round to 3 decimal places
+    : 0;
+  translatedAnswerDisplayText = translatedAnswerDisplayText.replace('$result', result);
```

Comment:
I'd like to try to group the logical concepts together more, so rather than setting up the display text and then translating it in a couple of places, a single concept is handled in a single place. In this instance, you could either pull all of your code for the display text down here just as it is, or even change it up a bit as follows
```suggestion
  // could be inline, or could be a util
  const replaceVariablesInText = (rawText, variables) => {
    let text = rawText;
    Object.entries(variables).forEach((variableName, value) => {
      text = text.replace(variableName, result);
    });
    return text;
  });
  const translatedAnswerDisplayText = replaceVariablesInText(answerDisplayText, { ...values, $result: result });
```